### PR TITLE
fix(es/typescript): Support `ts_enum_is_mutable` with const enums

### PR DESCRIPTION
**Pull request by dsherret**
**Description:** Bug I introduced in https://github.com/swc-project/swc/pull/8115/files (it was not implemented correctly)

### CODE REVIEW

File: crates/swc_ecma_transforms_typescript/tests/strip.rs [hunk 1]
```diff
@@ -4581,5 +4586,7 @@ test!(
     })(D || (D = {}));
     D.A = 5;
     console.log(D.A);
+    var E;
```
Comment (dsherret) [inline crates/swc_ecma_transforms_typescript/tests/strip.rs:4589]: Improvement for swc in the future might be to not emit this for const enums.

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_ecma_transforms_typescript

File: crates/swc_ecma_transforms_typescript/tests/strip.rs [hunk 2]
```diff
@@ -4567,19 +4567,57 @@ test!(
...
+    console.log(G.A);
+    var H;
+    (function(H) {
+        H[H["A"] = 2] = "A";
+    })(H || (H = {}));
+    var I;
+    (function(I) {
+        I[I["A"] = H.A] = "A";
+    })(I || (I = {}));
+    console.log(I.A);
```
Comment (dsherret) [inline crates/swc_ecma_transforms_typescript/tests/strip.rs:4621]: This is not ideal, but it's ok output. It's a rare edge case for someone to use an enum in a const enum.

**Review by kdy1 — APPROVED** [review]
